misc/metrics/src/protocol_stack.rs

```diff
@@ -0,0 +1,86 @@
+use libp2p_core::multiaddr::{Multiaddr, Protocol};
+use prometheus_client::encoding::text::Encode;
+
+#[derive(Encode, Hash, Clone, Eq, PartialEq)]
+pub struct Label {
```

Comment:
```suggestion
pub struct Labels {
```
Nit pick, a `struct` is a set of labels.